clouddq/integration/bigquery/bigquery_client.py

```diff
@@ -208,3 +208,61 @@ def execute_query(
         )
 
         return query_job
+
+    def get_table_schema(self, table: str, project_id: str = None) -> dict:
+
+        client = self.get_connection(project_id=project_id)
+        try:
+            table_ref = client.get_table(table)
+        except KeyError as error:
+            raise KeyError(f"\n\nInput table `{table}` is not valid.\n{error}")
+        columns = {}
+        for column in table_ref.schema:
+            column_configs = {
+                "name": column.name,
+                "type": column.field_type,
+                "mode": column.mode,
+                "data_type": column.field_type,
+            }
+            columns[column.name.upper()] = column_configs
+        table_type = table_ref.table_type
+        logger.debug(f"Table Type is : {table_type}")
+        table_partitioning_type = table_ref.partitioning_type
+        logger.debug(f"Table Partitioning Type is : {table_partitioning_type}")
+        if table_partitioning_type:
+            partition_fields = []
+            time_partitioning = table_ref.time_partitioning
+            if time_partitioning:
+                partition_field = {}
+                time_partitioning_type = time_partitioning.type_
+                time_partitioning_field = time_partitioning.field
+                if not time_partitioning_field:
+                    partition_fields = "_PARTITIONTIME"
+                else:
+                    partition_field["name"] = time_partitioning_field
+                    partition_field["type"] = columns[time_partitioning_field.upper()][
+                        "type"
+                    ]
+                    partition_field["partitioning_type"] = time_partitioning_type
+                    partition_fields.append(partition_field)
+
+            range_partitioning = table_ref.range_partitioning
+            if range_partitioning:
+                raise NotImplementedError(
+                    logger.debug(
```

Comment:
Should this be debug level logging or higher?

Reply:
This should be only message for NotImplementedError, logger is not needed. Thanks for catching this.